### ISSUE TITLE
Azure Selection of Zones Leads to Error for Machine Creation

### ISSUE THREAD

**Issue by toschneck** [kubermatic/kubermatic#6557]
**User Story**
Since 2.16.0 if I crate a new cluster or add a Machine Deployment, the selection of ANY zone leads to an error from the Azure API.
I also don't think a selection of multiple zones is really possible (or only if a dedicated Availability option of Azure is chosen before `Deploying an Availability Set to an Availability Zone isn’t supported.`).

As it is not working right now, it's needed to end-2-end test and adjust the user workflow (potential also on different azure datacenters as they may have different behavoirs)
![img.png](https://images.zenhubusercontent.com/5b6454d031c5fe7463837d8b/ae038ca1-3c3f-46a4-b6f5-208475053ad5)

```
E0210 13:58:20.785581       1 machine_controller.go:360] Failed to reconcile machine "azure-test-node-pool-588db5c476-2fkbf": failed to create machine at cloudprovider, due to failed to create public IP: failed to create public IP address: network.PublicIPAddressesClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="ResourceCannotHaveMultipleZonesSpecified" Message="Resource /subscriptions/528474a9-8cc6-46c8-9b74-1dbc31c4bb7f/resourceGroups/kubernetes-dv6hdv29rm/providers/Microsoft.Network/publicIPAddresses/azure-test-node-pool-588db5c476-2fkbf-netiface-pubip has 3 zones specified. Only one zone can be specified for this resource." Details=[]


	failed to create machine at cloudprovider, due to trying to create a VM: compute.VirtualMachinesClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="BadRequest" Message="Virtual Machine cannot be created because both Availability Zone and Availability Set were specified. Deploying an Availability Set to an Availability Zone isn’t supported."
```
**Acceptance criteria**
- [ ] user could create azure cluster as it is expected

**Comment by toschneck** [comment]
test at run-2 with latest 2.16.0 release

**Comment by zreigz** [comment]
@moadqassem
I'm not sure but it looks like the machine controller problem. It allows to pass the list of the zones:
https://github.com/kubermatic/machine-controller/blob/master/pkg/cloudprovider/provider/azure/types/types.go#L39

We just reflected this in the API and UI. Please investigate this issue and let us know if we can do something in the API or in the UI.
Unfortunately, I can't discover from Azure client which instances can work with multiple zones.

In the worst scenario, we can use the radio button in the UI to disable multi-selection

**Comment by moadqassem** [comment]
@zreigz Well, the azure client actually accepts list of zones and the the machine spec we only forwards that: https://github.com/Azure/azure-sdk-for-go/blob/master/services/network/mgmt/2018-06-01/network/models.go#L12592

**Comment by zreigz** [comment]
@moadqassem  I think it's matter how the VMs are created: single instances or inside an availability set: https://docs.microsoft.com/en-us/azure/virtual-machines/windows/tutorial-availability-sets

If as s single instance we can pass only one zone.

**Comment by zreigz** [comment]
failed to create machine at cloudprovider, due to failed to create public IP: failed to create public IP address: network.PublicIPAddressesClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="ResourceCannotHaveMultipleZonesSpecified" Message="Resource /subscriptions/528474a9-8cc6-46c8-9b74-1dbc31c4bb7f/resourceGroups/kubernetes-6p5sjj59r8/providers/Microsoft.Network/publicIPAddresses/wizardly-joliot-5d794d46b4-v7tgm-netiface-pubip has 3 zones specified. Only one zone can be specified for this resource." Details=[]

**Comment by moadqassem** [comment]
@zreigz The problem is we really can't this detail without digging into the api. I can investigate more about it.

**Comment by toschneck** [comment]
@moadqassem / @zreigz 
Me and @mfahlandt did take a look into the azure client and basically the implementation there is not correct to the internal system. You could EVERYTIME only assign one zone. Also the terraform porvider does ist wrong, but anyway for users who are not as deep (as we see in the discussion) this is not a nice behavior. 

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/virtual_machine#zones
>(Optional) A list of a single item of the Availability Zone which the Virtual Machine should be allocated in.

My personal best solution would be to change it every where to (ensured by validation):
1. just one field with only one selection of the availabilty zones are possible
2. OR the AvailabilitySet get choosen/created

Both options needs to be available for users
- Avialability Set -> Azure manages the spread of VMs
- Dedicated Zones -> User spezifies the target Zone

**Comment by shaase-ctrl** [comment]
Disable multi selection in UI, only allow for one zone.

**Comment by moadqassem** [comment]
@shaase-ctrl, @toschneck and @zreigz  The issue actually isn't completely in the UI/API. It is in the seed controller manager. The problem is, seed controller creates an availability set for the user cluster and attach it to the machines unconditionally. So in another words, av sets are on a cluster level(resource group level) and az's are on a machine deployment level. So you can't deploy a vm in an az if your cluster is attached to an av set. And since we assign an av set to cluster automatically on cluster creation, it will be impossible to create a vm in an az.

**Comment by moadqassem** [comment]
https://github.com/kubermatic/kubermatic/issues/7261

**Comment by rastislavs** [comment]
fixed in KKP v2.18
